solr/core/src/test/org/apache/solr/search/facet/TestJsonFacetRefinement.java

```diff
@@ -555,14 +556,14 @@ public void testPrelimSortingWithRefinement() throws Exception {
                          + "}")
                   , "facets=={ count: "+id+","
                   + "  cat_1:{ buckets:[ "
-                  + "            {val:X,count:4}," // index desc
-                  + "            {val:B,count:1}," 
-                  + "            {val:A,count:1}," 
+                  + "            {val:Z,count:2}," // index desc
+                  + "            {val:Y,count:5},"
+                  + "            {val:X,count:4},"
                   + "  ] },"
                   + "  cat_2:{ buckets:[ "
-                  + "            {val:X,count:4,sum_p:4.0}," // count desc
-                  + "            {val:A,count:1,sum_p:1.0}," // index order tie break
-                  + "            {val:B,count:1,sum_p:1.0},"
+                  + "            {val:C,count:6,sum_p:6.0}," // count desc
+                  + "            {val:Y,count:5,sum_p:5.0}," // index order tie break
+                  + "            {val:Z,count:2,sum_p:2.0},"
                   + "  ] }"
```

Comment:
The results in these tests are clearly better relative to request -- but probably would need to reformulate a little for upstream to truly show sorting behavior with regards to refinement.

Reply:
Yep yep, agreed these tests were trying to flex very specific behavior even if results are kind of bad. Will look into some updates to show new option behavior (once I also add option flag).